python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,413 @@
+
+        # Total duration calculation
+        total_duration = (
+            self.instrument_setup_time  # Initial setup time for the instrument
+            + target_flat_exptime * (n_flats)  # Time for taking all flats
+            + setup_time_per_image * (n_flats)  # Setup time p/image
+        )
+
+        metadata.duration = total_duration
+        metadata.instrument = self.get_instrument_name()
+        metadata.filter = self.get_instrument_filter()
+
+    def get_new_exptime(self, sky_counts, exp_time):
+        """Configure script components including camera.
+
+        Parameters
+        ----------
+        sky_counts : float
+            Counts in electrons of previous flat.
+        exp_time : float
+            Exposure time of previous flat
+
+        Returns
+        -------
+        float
+            Calculated new exposure time.
+        """
+
+        return self.config.target_sky_counts * exp_time / sky_counts
+
+    def get_target_radec(self, distance_from_sun=180, target_el=45, time=None):
+        """
+        Returns the RADEC of the target area of the sky that's an azimuth
+        `distance_from_sun` away from the Sun, given `elevation`,
+        and at a given `time`.
+
+        Parameters
+        ----------
+        distance_from_sun : float, (-180, 180)
+            The distance from the Sun in degrees.
+            Positive angles go towards the North.
+        elevation : float
+            Target elevation for Sky Flats.
+        time : datetime
+            The time for the calculation in UTC.
+        """
+
+        az_sun, el_sun = self.tcs.get_sun_azel()
+
+        target_az = (az_sun + distance_from_sun) % 360
+
+        target_radec = self.tcs.radec_from_azel(target_az, target_el)
+
+        return target_radec
+
+    def get_empty_field(self, target, radius=5):
+        """
+        Query the "Deep blank field catalogue : J/MNRAS/427/679" in Vizier.
+
+        Parameters
+        ----------
+        target : astropy.coordinates.SkyCoord
+            Sky coordinates near the field
+        radius : float
+            Search radius in degrees.
+
+        Reference
+        ---------
+        http://cdsarc.u-strasbg.fr/viz-bin/Cat?J/MNRAS/427/679
+        """
+        _table = Vizier.query_region(
+            catalog="J/MNRAS/427/679/blank_fld",
+            coordinates=target,
+            radius=radius * u.deg,
+        )
+
+        if len(_table) == 0:
+            self.log.info(
+                f"Could not find a field near {target} " f"within {radius} deg radius"
+            )
+            return None
+
+        _table = _table["J/MNRAS/427/679/blank_fld"]
+
+        coords = coordinates.SkyCoord(
+            ra=_table["RAJ2000"],
+            dec=_table["DEJ2000"],
+            unit=(u.hourangle, u.deg),
+            frame=coordinates.ICRS,
+        )
+
+        arg = target.separation(coords).argmin()
+
+        return coords[arg]
+
+    def confirm_sun_location(self):
```

Comment:
Consider renaming this method to `assert_sun_location` and then raise an `AssertionError` if the sun is not in the limits.